diagnostic.md

```diff
@@ -6,70 +6,73 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+get '/books/4', to 'books#show'
+get '/books/4', to 'books#show', as 'book'
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+this will execut the rake script with the command db:migrate which will run migrate on your database
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+rails g scaffold things will generate the model, the database for migrations, the controller and a test for things
+rails g controller things will generate a new controller named things
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+the serializer is responsible for definng the attributes associated with it so that the controller and model can access them
 ```
 
 What do we expect the command `Patient.all` to return?
 
 ```md
-// your response here
+this command should return all of the patients in the database
+using .all.each will load the entire table into memory and also iterate through each instance
 ```
 
 We use the constant `Patient` in the `PatientsController`, but where is it
 actually defined?
 
 ```md
-// your response here
+Patient is defined in the model, which the controller will use
 ```
 
 Why do our `model` files appear to be lacking so much code, when they're doing
 so much for us?
 
 ```md
-// your response here
+the model inherites from the ActiveRecord. Since the database is generated by rails, a lot of the code is already created
 ```
 
 What does typing `bin/rails dbconsole` in a directory that houses a rails app do for
 us?
 
 ```md
-// your response here
+this command finds the database we are currently working in and begins a session from the most recent point
 ```
 
 What does typing `rails db` in a directory that houses a rails app do for us?
 
 ```md
-// your response here
+this command will bring up the possible commands to be used by rails
```

Comment:
`rails db` opens up a PostgreSQL shell connected to the database that we're working with.